pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -85,11 +90,35 @@ type event struct {
 	oldObj interface{}
 }
 
+type concurrentUIDToNode struct {
+	*sync.RWMutex
+	uidToNode map[types.UID]*node
+}
+
+func (m *concurrentUIDToNode) Write(uid types.UID, node *node) {
```

Comment:
uid can be gotten from the node itself, I'd suggest you don't take it as a parameter. Same for below function.